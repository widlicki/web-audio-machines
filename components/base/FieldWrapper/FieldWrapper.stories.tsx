import { ComponentMeta, ComponentStory } from '@storybook/react'
import FieldWrapper from './FieldWrapper'

export default {
  title: 'components/base/FieldWrapper',
  component: FieldWrapper,
} as ComponentMeta<typeof FieldWrapper>

const Template: ComponentStory<typeof FieldWrapper> = (args) => (
  <FieldWrapper {...args}>
    <h4>This is some FieldWrapper Content</h4>
  </FieldWrapper>
)

export const DefaultFieldWrapper = Template.bind({})
DefaultFieldWrapper.args = {
  title: 'FieldWrapper Title',
}
