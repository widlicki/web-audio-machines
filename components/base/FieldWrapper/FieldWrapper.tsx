import { FormControl, FormGroup, FormLabel } from '@mui/material'
import theme from '../../../styles/Theme'

export interface FieldWrapperProps {
  title: string
  children?: JSX.Element
}

const FieldWrapper = ({ title, children }: FieldWrapperProps) => {
  return (
    <FormControl
      component="fieldset"
      sx={{
        width: '100%',
        border: `${theme.border.size.sm} solid ${theme.border.color.normal}`,
        borderRadius: 1,
        marginTop: theme.spacing(-1),
      }}
    >
      <FormLabel
        component="legend"
        sx={{
          fontSize: '12px',
          marginLeft: theme.spacing(1),
          paddingLeft: theme.spacing(1),
          paddingRight: theme.spacing(1),
        }}
      >
        {title}
      </FormLabel>
      <FormGroup>{children}</FormGroup>
    </FormControl>
  )
}

export default FieldWrapper
