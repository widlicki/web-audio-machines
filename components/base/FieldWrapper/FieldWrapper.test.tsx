import { render } from '@testing-library/react'
import FieldWrapper from './FieldWrapper'
import { DefaultFieldWrapper } from './FieldWrapper.stories'

describe('FieldWrapper Stories', () => {
  new Map([['Default FieldWrapper', DefaultFieldWrapper]]).forEach((key) => {
    it(`renders ${key}`, () => {
      const component = (
        <FieldWrapper title="test">
          <h2>some content</h2>
        </FieldWrapper>
      )
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})
