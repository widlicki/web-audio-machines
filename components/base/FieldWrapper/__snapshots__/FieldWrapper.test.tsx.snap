// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWrapper Stories renders function () { [native code] } 1`] = `
<div>
  <fieldset
    class="MuiFormControl-root css-heqxz8-MuiFormControl-root"
  >
    <legend
      class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1gye1hj-MuiFormLabel-root"
    >
      test
    </legend>
    <div
      class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <h2>
        some content
      </h2>
    </div>
  </fieldset>
</div>
`;
